### ISSUE TITLE
Add Cake.Codecov

### ISSUE THREAD

**Issue by larzw** [cake-contrib/Cake.Recipe#105]
As we discussed in Slack, it would be nice to have Codecov added. I believe @AdmiringWorm said he'd look into it, but let me know if you need any help!

**Comment by AdmiringWorm** [comment]
Yup, I'm currently working on it now.

**Comment by gep13** [comment]
@larzw thanks for raising this, and huge thank you to @AdmiringWorm for looking into this!  Please let me know if you have any questions.

**Comment by AdmiringWorm** [comment]
@gep13 well, I got one.
How can I test the changes?
I don't have any repos available that work with Cake.Recipe.

**Comment by gep13** [comment]
@AdmiringWorm almost all the repos in the Cake-Contrib Organisation work with Cake.Recipe, so you could pick one of those.  Or, if you are feeling really brave, you could convert Cake.Codecov to use Cake.Recipe, and dogfood the Codecov functionality that you add into Cake.Recipe 😄

**Comment by AdmiringWorm** [comment]
@gep13 I was under the impression that Cake.Recipe didn't work with .NET Core?
If so, then it would be pretty much impossible to convert Cake.Codecov to use Cake.Recipe.

**Comment by gep13** [comment]
@AdmiringWorm ah, apologies, I hadn't realised that is was using .NET Core (hadn't actually started looking at the source code).  In which case, yes, Cake.Recipe currently can't help you here.  It is VERY close to being there, but not quite.  If you pick one of the other repos in this Organisation, and replace the necessary files in the `tools\Cake.Recipe` folder with the ones you have changed, and run the build, you should be able to test it out that way.

**Comment by AdmiringWorm** [comment]
@gep13 how are the cake files loaded when using the recipe?
I tried to find out how the coveralls.cake was loaded to make it similar to the codecov.cake, or are they perhaps auto-loaded through some kind of scripting?
Last thing that needs to be done (well before testing it and finding compile errors, that is)

**Comment by gep13** [comment]
@AdmiringWorm have a look here:

https://github.com/cake-contrib/Cake.Prca.PullRequests.Tfs/blob/develop/setup.cake

The main entry point for Cake.Recipe is referencing the NuGet Package that contains the .cake files that Cake.Recipe ships, i.e. the first line in the above setup.cake file.  Once that is done, Cake then takes over and internally does a `#load` of each file contained within the NuGet package.

The remainder of the setup.cake file is then configuring the project specific properties for Cake.Recipe, and then starting the process.

Does that make sense?

**Comment by AdmiringWorm** [comment]
@gep13 
It makes sense, but doesn't really answer my question.
I mean, how to tell Cake.Recipe to load the `codecov.cake` file (internally).
Is it loaded automatically by just adding the file, or do I need to add `#load ./codecov.cake` somewhere?

I'm guessing automatically by your sentence:
> Once that is done, Cake then takes over and internally does a #load of each file contained within the NuGet package

**Comment by gep13** [comment]
Gotcha! Yes.  By simply adding the codecov.cake in the Cake.Recipe\Content folder, it will automatically be loaded into the project that is using Cake.Recipe to run it's build.
